jerry-core/debugger/debugger.c

```diff
@@ -955,15 +968,23 @@ jerry_debugger_process_message (const uint8_t *recv_buffer_p, /**< pointer to th
       }
 
       jerry_debugger_uint8_data_t *eval_uint8_data_p;
-      size_t eval_data_size = sizeof (jerry_debugger_uint8_data_t) + eval_size;
+
+      size_t eval_data_size = sizeof (jerry_debugger_uint8_data_t) + eval_size + sizeof (uint32_t);
```

Comment:
Why don't you simply copy all data? There is no need to copy chain_index separately, since it is also part of the message payload. This would simplify the code here a lot.